uber/site_sections/preregistration.py

```diff
@@ -162,7 +162,12 @@ def form(self, session, message='', edit_id=None, **params):
                 else:
                     target = group if group.badges else attendee
                     track_type = c.EDITED_PREREG if target.id in self.unpaid_preregs else c.UNPAID_PREREG
-                    self.unpaid_preregs[target.id] = to_sessionized(attendee, group)
+                    if attendee.paid == c.NEED_NOT_PAY:
```

Comment:
For what it's worth, the cart system is still in effect even with this change. I have verified the behavior locally as you describe.

1. Attendee fills out their badge & uses a promo code that makes it free
2. The confirmation page has a "Confirm Registration" button and a "Add Another Registration" button
3. Clicking the "Confirm Registration" actually processes the badge and adds the attendee to the system

So the cart system is still in full effect. I think what's happening is that `attendee.paid` is not set until the attendee is saved in the database, so the `if attendee.paid == c.NEED_NOT_PAY:` check will always fail.

That being said, I'll revert this change because I don't think it will ever actually do anything anyway.